Lesson4 hw

iourilitv opened the pull request.

iourilitv committed:
Solving the tasks of lesson 4 hw.
Completed a transferring from jsp/jstl to jsf.
All works correctly.

iourilitv committed:
Solving the tasks of lesson 4 hw.
Created the following entities: Category, User, Order, OrderItems.
Created .xhtml files for them excluding the OrderItems.
Added categoryId into Product and re arranged whole environment for it.
All works correctly.

iourilitv committed:
Completed all tasks of lesson 4 hw.
All works correctly.

iourilitv committed:
Deleted Test.
All works correctly.

iourilitv committed:
Just update TODO.
All works correctly.

usharik commented on lesson4-app/src/main/java/gb/lesson4/entities/Category.java:
```diff
@@ -0,0 +1,71 @@
+//TODO В index.xhtml при этой аннотации в списке товаров(на пустом) все елементы выводятся с пустыми полями!
+// Это из-за того, что локальная переменная и бин(сущность) называются одинаково.
+// Решение: просто переименовать локальную переменную(var) в index.xhtml или не использовать @RequestScoped для сущностей.
+public class Category {
```
Про валидацию поговорим на уроке

iourilitv committed:
Updating lesson4 hw.
Inactivated @Named and added @NotEmpty annotations in Category.
Added Categories into the navigation.
All works correctly.

iourilitv committed:
Updating lesson4 hw.
Updated ProductController and CategoryController, index.xhtml and categories.xhtml to fix the "Triple request to repository to get a List of entities" problem.
All works correctly.
